src/scripts/main.js

```diff
@@ -1,3 +1,18 @@
 'use strict';
 
-// write code here
+document.querySelectorAll('th').forEach((header) => {
+  header.addEventListener('click', () => {
+    const table = header.closest('table');
+    const columnIndex = Array.from(header.parentNode.children).indexOf(header);
+    const rows = Array.from(table.querySelectorAll('tr')).slice(1);
+
+    rows.sort((a, b) => {
+      const cellA = a.children[columnIndex].textContent;
+      const cellB = b.children[columnIndex].textContent;
+
+      return cellA.localeCompare(cellB, undefined, { numeric: true });
```

Comment:
The sorting logic uses `localeCompare` with the `numeric: true` option, which is generally correct for sorting numbers. However, ensure that the cell contents are indeed numeric strings. If the cells contain mixed data types or non-numeric strings, this could lead to unexpected sorting results. Consider parsing the cell contents to numbers if you expect purely numeric data.